op-e2e/config/init.go

```diff
@@ -328,8 +328,7 @@ func initAllocType(root string, allocType AllocType) {
 			l2Alloc[mode] = st.Chains[0].Allocs.Data
 			mtx.Unlock()
 
-			// This needs to be updated whenever the latest hardfork is changed.
-			if mode == genesis.L2AllocsGranite {
+			if mode == genesis.LatestHF {
 				dc, err := inspect.DeployConfig(st, intent.Chains[0].ID)
 				if err != nil {
 					panic(fmt.Errorf("failed to inspect deploy config: %w", err))
```

Comment:
I currently don't understand what this branch is actually about. We should clarify, maybe @mslipper knows?

Reply:
Basically, `l2AllocsByType` is two dimensional: `map[AllocType]map[L2AllocsMode]xxx`, but other maps are 1 dimensional, like `map[AllocType]yyy`, so one of them modes is picked as the canonical one.